components/provider_cldr/src/transform/timezones.rs

```diff
@@ -0,0 +1,401 @@
+
+impl<'d> IterableDataProvider<'d> for TimeZonesProvider<'d> {
+    fn supported_options_for_key(
+        &self,
+        _resc_key: &ResourceKey,
+    ) -> Result<Box<dyn Iterator<Item = ResourceOptions>>, DataError> {
+        let list: Vec<ResourceOptions> = self
+            .data
+            .iter()
+            .map(|(l, _)| ResourceOptions {
+                variant: None,
+                langid: Some(l.langid.clone()),
+            })
+            .collect();
+        Ok(Box::new(list.into_iter()))
+    }
+}
+/// Serde structs for the CLDR JSON zones files.
+pub(self) mod cldr_json {
+    use crate::cldr_langid::CldrLangID;
+    use serde::Deserialize;
+    use std::borrow::Cow;
+    //use litemap::LiteMap;
+    type LiteMap<K, V> = std::collections::BTreeMap<K, V>;
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    pub struct MetaZone {
+        pub long: Option<ZoneFormat>,
+        pub short: Option<ZoneFormat>,
+    }
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    pub struct MetaZones(pub LiteMap<Cow<'static, str>, MetaZone>);
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    pub struct ZoneFormat {
+        pub generic: Option<Cow<'static, str>>,
+        pub standard: Option<Cow<'static, str>>,
+        pub daylight: Option<Cow<'static, str>>,
+    }
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    pub struct LocationWithExemplarCity {
+        pub long: Option<ZoneFormat>,
+        pub short: Option<ZoneFormat>,
+        #[serde(rename = "exemplarCity")]
+        pub exemplar_city: Cow<'static, str>,
+    }
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    pub struct LocationWithShort {
+        pub long: Option<ZoneFormat>,
+        pub short: ZoneFormat,
+        #[serde(rename = "exemplarCity")]
+        pub exemplar_city: Option<Cow<'static, str>>,
+    }
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    pub struct LocationWithLong {
+        pub long: ZoneFormat,
+        pub short: Option<ZoneFormat>,
+        #[serde(rename = "exemplarCity")]
+        pub exemplar_city: Option<Cow<'static, str>>,
+    }
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    #[serde(untagged)]
+    pub enum Location {
+        LocationE(LocationWithExemplarCity),
+        LocationL(LocationWithLong),
+        LocationS(LocationWithShort),
+    }
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    #[serde(untagged)]
+    pub enum LocationOrSubRegion {
+        Location(Location),
+        SubRegion(LiteMap<Cow<'static, str>, Location>),
+    }
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    pub struct Region(pub LiteMap<Cow<'static, str>, LocationOrSubRegion>);
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    pub struct Zones(pub LiteMap<Cow<'static, str>, Region>);
+
+    #[derive(PartialEq, Debug, Clone, Deserialize)]
+    pub struct TimeZoneNames {
+        #[serde(rename = "hourFormat")]
+        pub hour_format: Cow<'static, str>,
```

Comment:
There's not much point in using Cows for the CLDR JSON serde struct, unfortunately.  It always get pulled from a reader, so data needs to be owned.  The good news is that this is typically run as an offline tool, so performance and heap allocations are less important.